Lib/ufo2ft/preProcessor.py

```diff
@@ -153,9 +156,13 @@ def initDefaultFilters(
 
         _init_explode_color_layer_glyphs_filter(self.ufo, filters)
 
-        # len(g) is the number of contours, so we include the all glyphs
-        # that have both components and at least one contour
-        filters.append(DecomposeComponentsFilter(include=lambda g: len(g)))
+        if flattenComponents:
+            from ufo2ft.filters.flattenComponents import FlattenComponentsFilter
+            filters.append(FlattenComponentsFilter())
+        else:
+            # len(g) is the number of contours, so we include the all glyphs
+            # that have both components and at least one contour
+            filters.append(DecomposeComponentsFilter(include=lambda g: len(g)))
```

Comment:
I think we still want to decompose mixed composite+contour glyphs regardless of whether we are also flattening nested components, it's not an if.. else..., but an if... if...  -- no?

Reply:
Oh, oops, yes, that would be best.

Reply:
Actually it seems like (from the tests) it doesn't make any difference. Does Flatten already do the work of Decompose?

Reply:
in theory no. One un-nests components, leaving behind composite glyphs with depth 1, the other removes components altogether leaving behind simple contour glyphs

Reply:
OK, then we'll do Flatten then Decompose, to be on the safe side.

Reply:
although.. if you have a mixed component+contour glyph, you really don't care about the components being nested or not. you just want to decompose. I think it's preferable to decompose mixed glyphs first, so you don't do extra work in the optional flatten filter.